### ISSUE TITLE
When I view no permission of repo should not be an 404 error.

### ISSUE THREAD

**Issue by zoujiaqing** [gogits/gogs#838]
need set an option: [show error] or [to login]?

**Comment by hickscorp** [comment]
I disagree. If a user is not logged in, trying to access a repository that exists privately should always return a 404. Otherwise, it's a way to know what users have access to what (Security flaw).

Eg when I am not logged in into Github, and try to access a private repository, the answer is a 404 as well. I would host any private repository on Github if it was different.
